lib/player.js

```diff
@@ -1822,9 +1822,18 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
           this.configure('manifest.disableVideo', true);
         }
 
+        // Init DRM engine if it's not created yet (happens on polyfilled EME).
+        if (!preloadManager.getDrmEngine()) {
+          await mutexWrapOperation(async () => {
+            preloadManager.initializeDrm();
+            await preloadManager.waitForDrmEngine();
```

Comment:
If I understand the code correctly, this can be simplified to

```suggestion
            await preloadManager.initializeDrm();
```
And you can remove waitForDrmEngine and drmEnginePromise_